### ISSUE TITLE
clarification of weak validators

### ISSUE THREAD

**Issue by reschke** [httpwg/http-core#163]
See <https://www.rfc-editor.org/errata/eid5236>

**Comment by reschke** [comment]
Mailing list thread: <https://lists.w3.org/Archives/Public/ietf-http-wg/2018JanMar/0041.html>

**Comment by mnot** [comment]
Not sure any change is necessary here; errata was rejected.

**Comment by tfpauly** [comment]
Discussed in Singapore; agreed to close this with no action, but @martinthomson offered to check if the text around validators could be clarified.

**Comment by martinthomson** [comment]
Just to close the loop on this, the document is entirely precise and perfectly obtuse at the same time.

It says that the validator applies to the selected representation.  So following the reference to Section 6 I find that the definition of selected representation correctly captures all the nuance, but fails to provide the easy cognitive shortcut of "selected representation =~ message body".  It manages to effectively obfuscate things by not clearly distinguishing between serialization and the abstract concept of what the resource state is.

I believe that selected representation is the specific projection of the platonic resource into bytes.  However, it doesn't really *say* that, it says that you interpret "bits" through the dual filters of content type and content coding.  Using "bits" is unfortunate.  XML schema has the dual notions of "value space" (the abstract form) and "lexical space" (the encoded form) and is very careful about being clear about which applies in which context.  This is not clear in the same way.

For instance, the idea that a representation is an abstraction (as stated) is something I find quite confusing.  The selected representation is a concrete form, expressed as a sequence of bytes.  That representation might be conveyed on the wire (a GET response sometimes, a PUT request) or it might be acted on or referred to.  Some methods operate exclusively on the bytes of a selected representation, even if those bytes don't hit the wire.

In short, the spec is correct, but it doesn't do a great job of teaching the protocol.

**Comment by mnot** [comment]
As that's a more general issue, closing in favour of #294.

**Comment by reschke** [comment]
So should we reject https://www.rfc-editor.org/errata/eid5236 then?
